File: leetcode/74-search-a-2d-matrix/search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m = matrix.size();
        int n = matrix[0].size();

        // if (n == 0 || )

        int l = 0;
        int h = (m*n)-1;

        while (l <= h){
            int mid = l+(h-l)/2;

            int k = mid/n;
            int g = mid%n;
            
            if (matrix[k][g] == target){
                return true;
            }else if (matrix[k][g] > target){
                h = mid-1;
            }else {
                l = mid+1;
            }
        }

        return false;
    }
};

File: leetcode/74-search-a-2d-matrix/search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        a = len (matrix[0])
        b = len (matrix)

        l = 0
        r = a*b -1

        while l<=r :

            mid = (l+r) // 2

            i = mid // a
            j = mid % a

            if matrix[i][j] == target:
                return True

            if matrix[i][j] > target:
                r = mid -1 

            else :
                l = mid + 1


        return False